boards: arduino: uno_r4: Add ADC, PWM, I2C, SPI, and TRNG

soburi opened the pull request:
Add support for basic peripherals.

note: I don't have a GY271 sensor module, so I haven't tested the i2c on an actual device or added support to yaml.
Some devices worked, so I don't think it's a problem.
I'll rebase after https://github.com/zephyrproject-rtos/zephyr/pull/85252 merged.

soburi committed:
boards: arduino: uno_r4: Add ADC, PWM, I2C, SPI, and TRNG

Add support for basic peripherals.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

yasinustunerg commented:
I think, 2 different issues are combined into 1 PR. One of them removing deprecated drivers (or configurations according to commit message) and adding new supported peripherals for some boards.

You may divide into 2 PRs. 
I made this request by referencing this [page](https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html)

soburi commented:
@fabiobaltieri 

I checked again and fixed the problem I found.
(Changed SPI of uno_r4_wifi to SPI0, changed reference voltage of adc_dt sample to 5V)

Could you check again, please?

thaoluonguw commented on tests/drivers/spi/spi_loopback/boards/arduino_uno_r4_r7fa4m1ab3cfm_minima.overlay:
```diff
@@ -0,0 +1,19 @@
+		compatible = "test-spi-loopback-slow";
+		reg = <0>;
+		spi-max-frequency = <2000000>;
+	};
+	fast@0 {
```
I think that it should be new line between these 2 lines.

soburi replied:
Fixed it.

soburi commented on tests/drivers/spi/spi_loopback/boards/arduino_uno_r4_r7fa4m1ab3cfm_minima.overlay:
```diff
@@ -0,0 +1,19 @@
+&spi1 {
+	rx-dtc;
+	tx-dtc;
+	slow@0 {
```
Fixed it.
